Formula/pocl.rb

```diff
@@ -0,0 +1,51 @@
+class Pocl < Formula
+  desc "Portable Computing Language"
+  homepage "http://portablecl.org"
+  url "http://portablecl.org/downloads/pocl-3.0.tar.gz"
+  sha256 "a3fd3889ef7854b90b8e4c7899c5de48b7494bf770e39fba5ad268a5cbcc719d"
+  license "MIT"
+  head "https://github.com/pocl/pocl.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "opencl-headers" => :build
+  depends_on "pkg-config" => :build
+  depends_on "hwloc"
+  depends_on "llvm"
+  depends_on "ocl-icd"
+
+  on_linux do
+    depends_on "gcc"
+  end
+
+  fails_with gcc: "5" # LLVM is built with GCC
+
+  def install
+    args = %W[
+      -DCMAKE_INSTALL_RPATH=#{lib};#{lib}/pocl
+      -DENABLE_EXAMPLES=OFF
+      -DENABLE_TESTS=OFF
+      -DLLVM_BINDIR=#{Formula["llvm"].opt_bin}
+      -DPOCL_INSTALL_ICD_VENDORDIR=#{prefix}/etc/OpenCL/vendors
```

Comment:
It looks like `#{prefix}/etc` is automatically symlinked to `#{etc}` via the Cellar path. Not sure how Homebrew handles this on updates (i.e. is there a broken symlink to Cellar path of previous version?)

I did try switching to `libexec` + `opt_libexec` in `install_symlink`, but it still uses the real Cellar path which is due to how `install_symlink` is implemented.

Looks like this needs a manual `ln_s` to use `opt...` paths.

Reply:
Based on some quick observations after a few more attempts:
- `#{prefix}/etc` is automatically symlinked to `#{etc}`
- Manually symlinking `#{opt_prefix}/etc` to `#{etc}` causes `brew link` to fail due to above.
- Manually symlinking `#{prefix}/etc` to `#{etc}` is skipped.
- `(etc/...).install_symlink` and related logic (like above automatic symlink) seem to overwrite files on updates. Although this may work right now, I didn't see any documentation on this and behavior doesn't match `(etc/...).install`, so wasn't sure how reliable.
- Retrying `libexec` + `opt_libexec` did seem to work.